src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/TripPatternForDate.java

```diff
@@ -176,6 +176,10 @@ public TripPatternForDate newWithFilteredTripTimes(Predicate<TripTimes> filter)
       return null;
     }
 
+    if (hasFrequencies() && filteredTripTimes.isEmpty() && filteredFrequencies.isEmpty()) {
```

Comment:
Should we extract the result of `hasFrequencies()` into a local variable?

Reply:
Potentially also use it to short-circuit parts of this function

Reply:
I've extracted it into a variable.

However, I'm not sure if I can short-circuit even more aggressively. Isn't that already the case? The only thing that I could skip would be the filtering of the frequencyEntries but in most cases that is an empty list. Is it worth it the extra complexity?

I'm happy to have a quick pairing session if you have a good idea.